pkg/api/pod/util_test.go

```diff
@@ -1187,50 +1187,59 @@ func TestDropProbeGracePeriod(t *testing.T) {
-			if newPod == nil {
-				continue
-			}
+	//enabled := true
+	for _, enabled := range []bool{true, false} {
+		for _, oldPodInfo := range podInfo {
+			for _, newPodInfo := range podInfo {
 
-			t.Run(fmt.Sprintf("feature enabled=%v, old pod %v, new pod %v", enabled, oldPodInfo.description, newPodInfo.description), func(t *testing.T) {
+				oldPodHasGracePeriod, oldPod := newPodInfo.hasGracePeriod, oldPodInfo.pod()
 
-				var oldPodSpec *api.PodSpec
-				if oldPod != nil {
-					oldPodSpec = &oldPod.Spec
-				}
-				dropDisabledFields(&newPod.Spec, nil, oldPodSpec, nil)
+				newPodHasGracePeriod, newPod := newPodInfo.hasGracePeriod, newPodInfo.pod()
 
-				// old pod should never be changed
-				if !reflect.DeepEqual(oldPod, oldPodInfo.pod()) {
-					t.Errorf("old pod changed: %v", cmp.Diff(oldPod, oldPodInfo.pod()))
+				if newPod == nil {
+					continue
 				}
 
-				switch {
-				case enabled || oldPodHasGracePeriod:
-					// new pod should not be changed if the feature is enabled, or if the old pod had subpaths
-					if !reflect.DeepEqual(newPod, newPodInfo.pod()) {
-						t.Errorf("new pod changed: %v", cmp.Diff(newPod, newPodInfo.pod()))
-					}
-				case newPodHasGracePeriod:
-					// new pod should be changed
-					if reflect.DeepEqual(newPod, newPodInfo.pod()) {
-						t.Errorf("new pod was not changed")
+				t.Run(fmt.Sprintf("feature enabled=%v, old pod %v, new pod %v", enabled, oldPodInfo.description, newPodInfo.description), func(t *testing.T) {
+
+					defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.ProbeTerminationGracePeriod, enabled)()
+
+					var oldPodSpec *api.PodSpec
+					if oldPod != nil {
+						oldPodSpec = &oldPod.Spec
 					}
-					// new pod should not have subpaths
-					if !reflect.DeepEqual(newPod, podWithoutProbeGracePeriod()) {
-						t.Errorf("new pod had probe-level terminationGracePeriod: %v", cmp.Diff(newPod, podWithoutProbeGracePeriod()))
+					dropDisabledFields(&newPod.Spec, nil, oldPodSpec, nil)
+
+					// old pod should never be changed
+					if !reflect.DeepEqual(oldPod, oldPodInfo.pod()) {
+						t.Errorf("old pod changed: %v", cmp.Diff(oldPod, oldPodInfo.pod()))
 					}
-				default:
-					// new pod should not need to be changed
-					if !reflect.DeepEqual(newPod, newPodInfo.pod()) {
-						t.Errorf("new pod changed: %v", cmp.Diff(newPod, newPodInfo.pod()))
+
+					switch {
+
+					case enabled || !oldPodHasGracePeriod:
+
+						// new pod should not be changed if the feature is enabled
+						if !reflect.DeepEqual(newPod, newPodInfo.pod()) {
+							t.Errorf("new pod changed: %v", cmp.Diff(newPod, newPodInfo.pod()))
+						}
+
+					case newPodHasGracePeriod:
+						// new pod should not have grace period if the feature gate is disabled
+						t.Errorf("new pod should not have grace period if the feature is disabled")
+
+					case reflect.DeepEqual(newPod, newPodInfo.pod()):
```

Comment:
This doesn't make sense to me as a case. We're unconditionally evaluating this if the first two cases don't happen, and if it's true, throwing an error. Whether they're equal is a test outcome, not a dev expectation. But, this is also opposite to the default case below, so I think this was just an oversight.

We need a case for `oldPodHasGracePeriod` to assert that the field was dropped. I'd suggest we repurpose this case for that.

Test matrix is

- Feature flag is on:
   - Pod has grace period: unchanged
   - Pod does not have grace period: unchanged
- Feature flag is off:
   - Pod has grace period: should be dropped
   - Pod does not have grace period: unchanged

Thus, I'd expect our cases to be:
- [x] `if enabled || !oldPodHasGracePeriod`: should be unchanged
- [x] Now we know it's disabled. So, we should check next, `if !enabled && newPodHasGracePeriod` and ensure the field is blanked out (note: it is probably clearer to write it the way I've written in this comment rather than trying to be clever with boolean logic)
- [ ] `if !enabled && oldHasGracePeriod`: should be dropped -- need a case for this still

I think we can keep the `default` case below as written unchanged.